cmd/postgres-operator/main.go

```diff
@@ -88,7 +88,8 @@ func main() {
 	done := make(chan bool, 1)
 	if upgradeCheckingEnabled {
 		log.Info("upgrade checking enabled")
-		go util.CheckForUpgradesScheduler(versionString, done)
+		go upgradecheck.CheckForUpgradesScheduler(done, versionString,
+			mgr.GetClient(), mgr.GetConfig(), isOpenshift(ctx, mgr.GetConfig()))
```

Comment:
:question: I have noticed that if there an attempt to use the client before the manager has started (i.e. before `mgr.Start()` is called a few lines down), and error will be thrown because the cache's for the manager/client have not yet synced.

Have you run into anything like this with your testing?  It seems like this would only _potentially_ be a concern for the first attempt to hit the endpoint when PGO starts.

Reply:
I have not seen that sort of error in testing this (either running the tests or running the program), but that might be just luck so far.

Actually, I just put in a sleep before the `mgr.Start` to try to force an error and that "successfully" errored; so I put in a WaitForCacheSync for-loop and that solved the error.

(I'm also beginning to think I should just pass the mgr in to retrieve the context, client, and cache, but that kind of obscures what we need, so I'll leave it for now, with some 🌱 for the future.)

Reply:
Sounds great, definitely on board with using `WaitForCacheSync()` here!